go/test/endtoend/tabletmanager/tablegc/tablegc_test.go

```diff
@@ -235,13 +264,28 @@ func dropTable(t *testing.T, tableName string) {
 	require.NoError(t, err)
 }
 
+func readMysqlVersion(t *testing.T) string {
+	query := `select @@version as ver`
+	rs, err := primaryTablet.VttabletProcess.QueryTablet(query, keyspaceName, true)
+	assert.NoError(t, err)
+	row := rs.Named().Row()
+	assert.NotNil(t, row)
+	version := row["ver"].ToString()
+	assert.NotEmpty(t, row)
+	return version
+}
+
+func isMySQL8() bool {
+	return strings.HasPrefix(mysqlVersion, "8.")
+}
+
+func TestMySQLVersion(t *testing.T) {
+	mysqlVersion = readMysqlVersion(t)
+}
```

Comment:
Don't have to do it in this PR, but I'd like to see us use https://github.com/vitessio/vitess/blob/main/go/vt/mysqlctl/capabilityset.go

The `isMySQL8` check will probably become more widely needed.

Reply:
Oooooh, that's interesting, I wasn't aware of `capabilities`. Definitely gonna do some work around that; I have _other_ stuff I've already pushed to `Conn` rather than `capabilities` such as "is fast drop table supported". Will do so in a different PR.